pkg/eks/api/api.go

```diff
@@ -41,22 +42,45 @@ var DefaultWaitTimeout = 20 * time.Minute
 // DefaultNodeCount defines the default number of nodes to be created
 const DefaultNodeCount = 2
 
-// ClusterProvider provides an interface with the needed AWS APIs
+// ClusterLocator is what identifies a cluster
+type ClusterLocator struct {
```

Comment:
I'm struggling with the terms `ClusterLocator` and `Locator`. I really like the fact we have separated out the fields that uniquely identify a cluster into a separate struct. And from this we can return the string representations etc. 
We are saying that `ClusterLocator` uniquely identifies a cluster and it just so happens that the identity is made up of a location part. I wonder if `ClusterIdentitity` or something similar better describes its purpose.

Reply:
I like `ClusterMeta`. Would it be worth doing that change or just take an action to change `ClusterLocator` in the future?

Reply:
Done.